packages/flutter_rx_bloc/lib/src/rx_bloc_multi_builder.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:rx_bloc/rx_bloc.dart';
+
+import 'base/stream_multi_builder_3.dart';
+import 'rx_bloc_provider.dart';
+
+/// region RxBlocMultiBuilder2
+
+/// RxBlocMultiBuilder2 is a Flutter widget which requires a [bloc] from the
+/// RxBloc ecosystem, a [builder], [state1] and [state2] function.
+/// RxBlocMultiBuilder2 handles building the widget in response to new states.
+/// RxBlocMultiBuilder2 is very similar to StreamBuilder but has a more simple
+/// API to reduce the amount of boilerplate code needed.
+///
+/// - The builder function will potentially be called many times and should be
+/// a pure function that returns a widget in response to the state.
+/// - The state function determines which exact state of the bloc will be used.
+/// - If the bloc parameter is omitted, RxBlocMultiBuilder2 will automatically
+/// perform a lookup using RxBlocProvider and the current BuildContext.
+class RxBlocMultiBuilder2<B extends RxBlocTypeBase, T1, T2>
+    extends StatelessWidget {
+  /// Default RxBlocBuilder constructor
+  const RxBlocMultiBuilder2({
```

Comment:
I updated the code for the multi builder instances, but I didn't include the lint rule, since it's going to require changes across quite a few files and the pull request is already getting pretty big.